### DISCUSSION TITLE
Want to disable notifications for a single line or a single section

### DISCUSSION THREAD

**Discussion by wb8tyw** [microsoft/pylance-release#1629] [Ideas]
While it may be ideal to fix all issues flagged, in many cases that is not practical, and a system is desired so that only notifications for new issues or changed code are flagged.

This is wanted when trying to modernize older code, or to deal with changes in rules to the code scanning code that finds new issues.

My current home project is converting an old application from python2/GTK-2/libxml to python3/gtk-3/lxml.

The first phase is to convert to gtk-3 and lxml, so while python3 is the eventual target, for now the code needs to work on python2 so that I can debug it.

It would take far to long to fix all of the existing issues that pylint and now pylance are flagging, so I have to triage them as to issues that can be quickly fixed, and issues that must be deferred for a later pass.

I only found out about pylance when it started flagging additional issues in the source code, some of which are not blockers to getting the conversion done.  Some issues will eventually be fixed by dropping the python2 support.

The PEP 484 specification for adding a # type: ignore is not taking into account that many of the lines that need the tags added are already at the maximum line length, and sometimes can not be easily refactored to shorter lines.

That feature of adding a "# type: ignore" to the end of a line is also not working correctly on the pylance bundled with Visual Studio Code 1.58.2.

Instead of operating on just that line, it is still operating on the entire file.  So disabling it for line where it is a false positive like using importing modules from "six.moves" disables it for other places where it might be flagging a real issue that got introduced with a change.

Other code analysis and lint tools that I work with use comments at the beginning of a block or a scope or
at just before a line to disable a check.  This works very well and is already a common convention.

Another convention that is used is to have the comments turn a check off only from the point where the comment is added and then a later comment can re-enable the check for the rest of the file.

**Comment by erictraut** [comment]
> That feature of adding a "# type: ignore" to the end of a line is also not working correctly on the pylance bundled with Visual Studio Code 1.58.2. Instead of operating on just that line, it is still operating on the entire file.

Sounds like a bug. Can you provide a code sample (or a public source file would work) that exhibits this behavior? If so, we can take a look at it.

**Reply by wb8tyw** [reply]
`from six.moves import range
from six.moves.configparser import NoOptionError # pyright: reportMissingModuleSource=false`

Just add these two lines for python2 to python3 compatibility.
Without the # pyright: ... line, both lines are flagged by that rule.

After adding that tag to the end of the either line, neither line is flagged by the rule.

**Reply by erictraut** [reply]
Ah, that's why. You're not using the comments correctly. If you want to silence all diagnostics on a single line, you need to use `# type: ignore` at the end of that line. The comment you're using is meant to control diagnostic reporting for the entire file. So it's working as designed.

**Reply by wb8tyw** [reply]
Yes, I can confirmt that.  I thought that "type" was a word to be replaced with what specifically was to be ignored.

**Reply by wb8tyw** [reply]
Since the imports from six are valid for python2 or python3, and are present on the system, why is pylance even flagging them?

**Reply by jakebailey** [reply]
See https://github.com/microsoft/pylance-release/issues/1343.

This is tricky because `six` creates the `moves` module at runtime, there's no source there. We need to report when the source is not there but stubs are so that we can tell people they haven't installed a package or are using the wrong interpreter.

I'd suggest voting/commenting on that issue if you hit this.
